src/axom/quest/DiscreteShape.cpp

```diff
@@ -0,0 +1,805 @@
+      const auto& tet = tets[iTet];
+      for(int i = 0; i < 4; ++i)
+      {
+        axom::IndexType iNode = iTet * 4 + i;
+        const auto& coords = tet[i];
+        nodeCoordsView[iNode][0] = coords[0];
+        nodeCoordsView[iNode][1] = coords[1];
+        nodeCoordsView[iNode][2] = coords[2];
+        connectivityView[iTet][i] = iNode;
+      }
+    });
+
+  TetMesh* tetMesh = nullptr;
+  if(m_sidreGroup != nullptr)
+  {
+    tetMesh = new TetMesh(3,
+                          axom::mint::CellType::TET,
+                          m_sidreGroup,
+                          nodeCoords.shape()[0],
+                          connectivity.shape()[0]);
+  }
+  else
+  {
+    tetMesh = new TetMesh(3,
+                          axom::mint::CellType::TET,
+                          nodeCoords.shape()[0],
+                          connectivity.shape()[0]);
+  }
+  tetMesh->appendNodes((double*)nodeCoords.data(), nodeCoords.shape()[0]);
+  tetMesh->appendCells(connectivity.data(), connectivity.shape()[0]);
+  m_meshRep.reset(tetMesh);
+
+  applyTransforms();
+}
+
+void DiscreteShape::createPlaneRepresentation()
+{
+  const axom::klee::Geometry& geometry = m_shape.getGeometry();
+
+  const auto& plane = geometry.getPlane();
+  // Generate a big bounding hex on the positive side of the plane.
+  /*
+    TODO: Use a specialized plane representation.  Plane is very
+    simple, but representing it as huge tets has 2 problems.
+    1. The shaper runs slow because the huge tets contains huge
+       numbers of cells, disbling attempts at divide-and-conquer
+       with the BVH.
+
+    2. When the tets are orders of magnitude larger then the
+       computational mesh, I've seen weird overlap computation
+       errors that affect certain resolutions and don't follow
+       any trends.  I think it's due to round-off, but I've not
+       verified.
+  */
+  axom::primal::Hexahedron<double, 3> boundingHex;
+  const double len = 1e2;  // Big enough to contain anticipated mesh,
+                           // but too big invites weird round-off errors.
+  // We should compute based on the mesh.
+  // clang-format off
+  boundingHex[0] = Point3D{0.0, -len, -len};
+  boundingHex[1] = Point3D{len, -len, -len};
+  boundingHex[2] = Point3D{len,  len, -len};
+  boundingHex[3] = Point3D{0.0,  len, -len};
+  boundingHex[4] = Point3D{0.0, -len,  len};
+  boundingHex[5] = Point3D{len, -len,  len};
+  boundingHex[6] = Point3D{len,  len,  len};
+  boundingHex[7] = Point3D{0.0,  len,  len};
+  // clang-format on
+  numerics::Matrix<double> rotate = vorAxisRotMatrix(plane.getNormal());
+  const auto translate = plane.getNormal() * plane.getOffset();
+  for(int i = 0; i < 8; ++i)
+  {
+    Point3D newCoords;
+    numerics::matrix_vector_multiply(rotate,
+                                     boundingHex[i].data(),
+                                     newCoords.data());
+    newCoords.array() += translate.array();
+    boundingHex[i].array() = newCoords.array();
+  }
```

Comment:
Please add a comment about what this block is doing. It looks like it's applying a rotation matrix to the bounding box to ensure it's covering the proper half-space.

Reply:
Done.